### PR TITLE
created a set of example configuration files for datadog agent

### PR DESCRIPTION
**Pull request by dogeahill**
### What does this PR do?

This change adds a directory with example `datadog-agent.yaml` configuration files

### Motivation

Frequently when implementing Datadog, a lot of time is spent `yaml` wrangling to get basic configurations set up. This process is tedious and error prone, and having a base set of configurations would simplify this process dramatically. Ideally it could ship with the agent when installed so that users could onboard datadog without having to parse a 3k plus line example 
yaml file.

### CODE REVIEW

**Comment by spencergilbert** [comment]
> Ideally it could ship with the agent when installed so that users could onboard datadog without having to parse a 3k plus line example yaml file.

This PR as-is won't include the example configurations in the distributed packages, I think we'd need to update the [omnibus](https://github.com/DataDog/datadog-agent/tree/main/omnibus) config to include this new folder, as well as where to put it when installed.

**Comment by dogeahill** [comment]
> > Ideally it could ship with the agent when installed so that users could onboard datadog without having to parse a 3k plus line example yaml file.
> 
> This PR as-is won't include the example configurations in the distributed packages, I think we'd need to update the [omnibus](https://github.com/DataDog/datadog-agent/tree/main/omnibus) config to include this new folder, as well as where to put it when installed.

Understood, that was more of a desired future state but the goal is just to have this available in this repository to offer simpler ways to configure the agent for initial onboarding.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+This is a collection of example `datadog-agent.yaml` files to get you started with Datadog. Consult the
+[config_template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) for a full list of configuration options. 
+
+To use these add your `api_key` and if necessary update the `site`. Add an `env` 
+tag to the `env:` key and any other required tags. If these parameters are set 
+with environment variables, they can be commented out. 
+
+Add any other configuration settings needed, then you can copy the file to `/etc/datadog-agent/datadog.yaml` 
+for Linux systemsor `%ProgramData%\Datadog\datadog.yaml` for Windows and restart the Datadog Agent.
```
Comment (spencergilbert) [inline examples/README.md:11]:
```suggestion
for Linux systems or `%ProgramData%\Datadog\datadog.yaml` for Windows and restart the Datadog Agent.  
```

File: examples/agent_otel.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+## metric, event, log, trace, and service check emitted by this Agent.  
+env:
+
+## The site of the Datadog intake to send Agent data to.
+site: datadoghq.com    
+
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+tags:
+  - example:changeme
+  - <TAG_KEY>:<TAG_VALUE>
```
Comment (spencergilbert) [inline examples/agent_otel.yaml:16]: **🥜 nitpick:** ‏ the other examples have this one line commented out

**Review by spencergilbert — APPROVED** [review]

File: examples/agent_apm_liveprocess.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+## The site of the Datadog intake to send Agent data to.
+site: datadoghq.com
+    
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+tags:
+  - example:changeme
+#  - <TAG_KEY>:<TAG_VALUE>
+
```
Comment (spencergilbert) [inline examples/agent_apm_liveprocess.yaml:19]:
```suggestion
```

File: examples/agent_apm_liveprocess.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+## Minimal configuration to enable Datadog to ship metrics, traces and Live Processes.
+
+## @env DD_API_KEY - string - required
+## The Datadog API key used by your Agent to submit metrics and events to Datadog.
+api_key:
+
+## The environment name where the agent is running. Attached in-app to every
+## metric, event, log, trace, and service check emitted by this Agent.  
+env:
+
```
Comment (spencergilbert) [inline examples/agent_apm_liveprocess.yaml:10]:
```suggestion
```

File: examples/agent_logs.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+## The environment name where the agent is running. Attached in-app to every
+## metric, event, log, trace, and service check emitted by this Agent.  
+env:
+
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+tags:
+  - example:changeme
+#  - <TAG_KEY>:<TAG_VALUE>
+
```
Comment (spencergilbert) [inline examples/agent_logs.yaml:18]:
```suggestion
```

File: examples/agent_logs_liveprocess.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+env:
+
+## The site of the Datadog intake to send Agent data to.
+site: datadoghq.com    
+
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+tags:
+  - example:changeme
+#  - <TAG_KEY>:<TAG_VALUE>
+
```
Comment (spencergilbert) [inline examples/agent_logs_liveprocess.yaml:18]:
```suggestion
```

**Pull request merged**